content/syllabus-written-assignments/file-directory-naming/_index.md

```diff
@@ -0,0 +1,28 @@
+---
+content_type: project
+flavours:
+  - markdown
+prerequisites:
+  hard:
+    - topics/file-and-directory-naming/general
+protect_main_branch: false
+ready: true
+submission_type: repo
+title: File and Directory Naming
+---
+
+This assessment is designed to not only assess your current knowledge but also provide insights into areas where you can further improve. It is an integral part of your learning journey, contributing to your growth as a capable and confident programmer.
+
+## Questions
+
+1. What is the recommended naming convention for files and directories in the programming language you are learning? Why is it important to follow this convention?
+
+2. What are some of the common mistakes that beginners make when naming files and directories? How can you avoid these mistakes?
+
+3. Explain the importance of using naming conventions when organizing files and directories. How do naming conventions help with locating, identifying, and retrieving information from a file system? Provide an example of a specific naming convention that you might use for categorizing different types of files.
```

Comment:
The questions are loaded. I think it would be good if each question had a single and not a multipart question because learners are likely to answer some parts and leave other parts for a question.

1. the question can be broken down into 2. one about naming conventions and the other about the important of naming conventions
2. I'm not sure if the second question is necessary if the answer is like just going to be "follow naming conventions"
3. This one can also be broken down further